### PR TITLE
fix: show error message when project initalization fail

### LINKED ISSUE #7221: cannot read Stacks of null during init when root stack creation fails
### Before opening, please confirm:

- [X] I have installed the latest version of the Amplify CLI (see above), and confirmed that the issue still persists.
- [X] I have [searched for duplicate or closed issues](https://github.com/aws-amplify/amplify-cli/issues?q=is%3Aissue+).
- [X] I have read the guide for [submitting bug reports](https://github.com/aws-amplify/amplify-cli/blob/master/CONTRIBUTING.md#bug-reports).
- [X] I have done my best to include a minimal, self-contained set of instructions for consistently reproducing the issue.

### How did you install the Amplify CLI?

yarn

### If applicable, what version of Node.js are you using?

_No response_

### Amplify CLI Version

latest

### What operating system are you using?

Mac

### Amplify Categories

Not applicable

### Amplify Commands

init

### Describe the bug

When bucket limit is reached or the root stack is creation is failed for some other reason, a `null` check is missing for `stackCreationData`. 

### Expected behavior

User friendly error message and proper handling of null value.

### Reproduction steps

init a project and fail the stack creation

### GraphQL schema(s)

<details>

```graphql
# Put schemas below this line


```

</details>

### Log output

<details>

```
# Put your logs below this line


```

</details>

### Additional information

_No response_

### PR DESCRIPTION
**Pull request by yuth**
#### Description of changes

The CLI throws ugly error if the initialization of project fails when the bucket limit is hit. This change shows list of errors instead of a stack trace
## Old error
```
ROLLBACK_COMPLETE amplify-initfail-dev-160404 AWS::CloudFormation::Stack Fri Apr 30 2021 16:04:20 GMT-0700 (Pacific Daylight Time)
⠏ Initializing project in the cloud...An error occurred when creating the CloudFormation stack
✖ Root stack creation failed
TypeError: Cannot read property 'Stacks' of null
    at processStackCreationData (/Users/yathiraj/.config/yarn/global/node_modules/amplify-provider-awscloudformation/src/initializer.js:108:44)
    at Object.run (/Users/yathiraj/.config/yarn/global/node_modules/amplify-provider-awscloudformation/src/initializer.js:80:7)
    at processTicksAndRejections (internal/process/task_queues.js:97:5)
```
## New error

```
⠧ Initializing project in the cloud...

Following resources failed

Resource Name: amplify-initfail-dev-155719-unauthRole (AWS::IAM::Role)
Event Type: create
Reason: Resource creation cancelled


Resource Name: amplify-initfail-dev-155719-authRole (AWS::IAM::Role)
Event Type: create
Reason: Resource creation cancelled


Resource Name: DeploymentBucket (AWS::S3::Bucket)
Event Type: create
Reason: You have attempted to create more buckets than allowed (Service: Amazon S3; Status Code: 400; Error Code: TooManyBuckets; Request ID: 9PE5RBXKCAHYN2ZC; S3 Extended Request ID: YXRhl55q/zkp8k23YWOR89/ypoPAE2ZVmMbXGTphHdL63MM1aO/zF1CrR1yi9rkoln29v/8qDuI=; Proxy: null)


✖ Root stack creation failed
``` 

#### Issue #, if available
fix #7221

#### Checklist

- [x] PR description included
- [x] `yarn test` passes
- [x] Tests are [changed or added](https://github.com/aws-amplify/amplify-cli/blob/master/CONTRIBUTING.md#tests)
- [ ] Relevant documentation is changed or added (and PR referenced)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Review by jhockett — APPROVED** [review]
LGTM

**Review by cjihrig — APPROVED** [review]

**Review by cjihrig — CHANGES_REQUESTED** [review]
It looks like there is a CI failure. I'm going to mark as changes requested so this doesn't get accidentally merged.

**Comment by yuth** [comment]
> It looks like there is a CI failure. I'm going to mark as changes requested so this doesn't get accidentally merged.

Tests are passing now

File: packages/amplify-provider-awscloudformation/src/__tests__/system-config-manager.test.ts [hunk 1]
```diff
@@ -2,6 +2,8 @@ import { $TSContext } from 'amplify-cli-core';
 import fs from 'fs-extra';
 import { getProfileCredentials, getProfiledAwsConfig } from '../system-config-manager';
 
+jest.setTimeout(15000);
```
Comment (cjihrig) [inline packages/amplify-provider-awscloudformation/src/__tests__/system-config-manager.test.ts:5]: Is this needed?

**Review by cjihrig — APPROVED** [review]

File: packages/amplify-provider-awscloudformation/src/__tests__/system-config-manager.test.ts (hunk 1 above)
Comment (yuth) [reply packages/amplify-provider-awscloudformation/src/__tests__/system-config-manager.test.ts:5]: Yes. The await otherwise times out

**Pull request merged**